internal/mode/static/nginx/config/generator.go

```diff
@@ -177,3 +215,98 @@ func generateCertBundle(id dataplane.CertBundleID, cert []byte) file.File {
 func generateCertBundleFileName(id dataplane.CertBundleID) string {
 	return filepath.Join(secretsFolder, string(id)+".crt")
 }
+
+type mgmtConf struct {
+	Endpoint            string
+	Resolver            string
+	DeploymentCtxFile   string
+	SkipVerify          bool
+	CACertExists        bool
+	ClientSSLCertExists bool
+}
+
+// generateMgmtFile generates the NGINX Plus configuration file for the mgmt block. As part of this,
+// it writes the deployment context file that is referenced in the mgmt block.
+func (g GeneratorImpl) generateMgmtFiles() []file.File {
```

Comment:
I would probably make this an executeFunc to stay consistent.

Reply:
Had to tweak slightly because we also now have to write secret files, so it's not a standard executeFunc (which assumes a Regular type file).